tests/resources/OVA-Util.robot

```diff
@@ -0,0 +1,90 @@
+
+${ova_target_vm_name}  vic-unified-ova-integration-test
+${ovftool_options}  --noSSLVerify --acceptAllEulas --name=${ova_target_vm_name} --diskMode=thin --powerOn --X:waitForIp --X:injectOvfEnv --X:enableHiddenProperties
+
+${ova_network_ip0}  10.17.109.207
+${ova_network_netmask0}  255.255.255.0
+${ova_network_gateway}  10.17.109.253
+${ova_network_dns}  10.118.81.1
+${ova_network_searchpath}  eng.vmware.com
+${ova_network_domain}  mrburns
+${ova_network_options}  --prop:network.ip0=${ova_network_ip0} --prop:network.netmask0=${ova_network_netmask0} --prop:network.gateway=${ova_network_gateway} --prop:network.DNS=${ova_network_dns} --prop:network.searchpath=${ova_network_searchpath} --prop:network.domain=${ova_network_domain}
+
+${ova_harbor_admin_password}  harbor-admin-passwd
+${ova_harbor_db_password}  harbor-db-passwd
+${ova_service_options}  --prop:harbor.admin_password=${ova_harbor_admin_password} --prop:harbor.db_password=${ova_harbor_db_password}
+
+${ova_options}  ${ovftool_options} ${ova_appliance_options} ${ova_service_options}
+${ova_options_with_network}  ${ova_options} ${ova_network_options} 
+
+${tls_not_disabled}  False
+
+# Requires vc credential for govc
+*** Keywords ***
+Deploy VIC-OVA To Test Server
+    [Arguments]  ${dhcp}=False  ${build}=False  ${user}=%{TEST_USERNAME}  ${password}=%{TEST_PASSWORD}  ${host}=%{TEST_URL}  ${datastore}=%{TEST_DATASTORE}  ${cluster}=%{TEST_RESOURCE}  ${datacenter}=%{TEST_DATACENTER}
+    Run Keyword if  ${build}  Build Unified OVA
+    ${rev}=  Run  git rev-parse --short HEAD
+    Set Test Variable  ${ova_path}  bin/vic-1.1.0-${rev}.ova
+
+    Log To Console  \nCleanup environment...
+    Run Keyword And Ignore Error  Run  GOVC_URL=${host} GOVC_USERNAME=${user} GOVC_PASSWORD=${password} GOVC_INSECURE=1 govc vm.destroy ${ova_target_vm_name}
+    Run Keyword And Ignore Error  Run  GOVC_URL=${host} GOVC_USERNAME=${user} GOVC_PASSWORD=${password} GOVC_INSECURE=1 govc object.destroy /${datacenter}/vm/${ova_target_vm_name}
+
+    Log To Console  \nStarting to deploy unified-ova to test server...
+    Run Keyword If  ${dhcp}  Log To Console  ovftool --datastore=${datastore} ${ova_options} ${ova_path} 'vi://${user}:${password}@${host}/${datacenter}/host/${cluster}'
+    ...  ELSE  Log To Console  ovftool --datastore=${datastore} ${ova_options_with_network} ${ova_path} 'vi://${user}:${password}@${host}/${datacenter}/host/${cluster}'
+    ${out}=  Run Keyword If  ${dhcp}  Run  ovftool --datastore=${datastore} ${ova_options} ${ova_path} 'vi://${user}:${password}@${host}/${datacenter}/host/${cluster}'
+    ...  ELSE  Run  ovftool --datastore=${datastore} ${ova_options_with_network} ${ova_path} 'vi://${user}:${password}@${host}/${datacenter}/host/${cluster}'
+    Log  ${out}
+
+    Log To Console  \n${out}
+    @{out}=  Split To Lines  ${out}
+    Should Contain  @{out}[-1]  Completed successfully 
+
+    Log To Console  \nUnified OVA is deployed successfully
+    Sleep  6s
```

Comment:
this is because govc may return empty query result on guest even though ovf deployment returns successfully. But you are right, the same principle applies. I'll change to a looped govc  with a maximal retry number